CONTRIBUTING.md

```diff
@@ -262,3 +273,72 @@ pre-commit run --files notebook1.ipynb notebook2.ipynb
 ```
 
 replacing `notebook1.ipynb` and `notebook2.ipynb` with any notebook(s) you have modified.
+
+## Maintainer guide
+
+We try to use automation to make the release process reliable, transparent and
+reproducible. This also helps us make releases more frequently.
+
+### Versioning
+
+We use `setuptools-scm` to source the version number from the git history
+automatically. At build time, the version number is determined from the git tag.
+
+shap uses a PEP 440-compliant versioning scheme of `MAJOR.MINOR.PATCH`. Like
+[numpy](numpy_versioning), shap does *not* use semantic versioning, and has
+never made a `major` release. Most releases increment `minor`, typically made
+every month or two. `patch` releases are sometimes made for any important
+bugfixes.
+
+Breaking changes are done with care, given that shap is a very popular package.
+When breaking changes are made, the PR should be tagged with the `BREAKING`
+label to ensure it is highlighted in the release notes. Deprecation cycless are
+used to mitigate the impact on downstream users.
+
+GitHub milestones can be used to track any actions that need to be completed for
+a given release, such as those relating to deprecation cycles.
+
+[numpy_versioning]: https://numpy.org/doc/stable/dev/depending_on_numpy.html
+
+
+### Making releases
+
+In the run-up to a release, create a GitHub issue for the release such as [[Meta
+issue] Release 0.43.0](https://github.com/shap/shap/issues/3289). This can be
+used to co-ordinate with other maintainers and agree to make a release.
+
+When a new GitHub [release](https://github.com/shap/shap/releases) is made, the
+wheels will be built and published to GitHub automatically by the `build_wheels`
+GitHub action. This workflow can also be triggered manually at any time to do a
+dry-run of cibuildwheel.
+
+Suggested release checklist:
+
+```
+- [ ] Dry-run cibuildwheel & test
+- [ ] Make GitHub release & tag
+- [ ] Confirm PyPI wheels published
+- [ ] Conda forge published
+```
+
+The conda package is managed in a separate repo:
+https://github.com/conda-forge/shap-feedstock . The conda-forge bot will
+automatically make a PR to this repo to update the conda package, typically
+within a few hours of the PyPSA package being published.
+
+### Release notes from PR labels
+
+Release notes can be automatically drafted by Github using the titles and labels
+of PRs that were merged since the previous release.The draft notes are
+configured in [.github/release.yml](.github/release.yml). See the [GitHub
+docs][auto_release_notes] for more info.
+
+It's helpful to assign labels such as `BREAKING`, `bug`, `enhancement` or
+`skip-changelog` to each PR, so that the change will show up in the notes under
+the right section. It also helps to ensure each PR has a descriptive name.
```

Comment:
Could we add a line here, what exactly collects the labels and puts it in the right section in the release notes? Is that a github bot or something else?

Or is all information contained in the auto_release_notes link you provided further down? Then maybe mentioning the link would suffice.

Reply:
Yes, lots more information about automatically generated release notes is in that link. Is the hyperlink above sufficient?

> See the [GitHubdocs][auto_release_notes] for more info.

I'll add a bit more text to explain how the labels are used, might as well be explicit I suppose.